src/Samples/SubModelSeq/Program.fs

```diff
@@ -13,177 +31,199 @@ module List =
         | a when a = j -> i
         | a -> a)
 
+  let swapWithNext i = swap i (i + 1)
+  let swapWithPrev i = swap i (i - 1)
+
+  let cons a ma = a :: ma
```

Comment:
```suggestion
  let cons x xs = x :: xs
```

Reply:
I prefer `a` over `x` since it matches the inferred type assigned by Visual Studio's IntelliSense and visible in the tooltips.

If the type were `Counter list`, then I would name it `counters`.  But when the (inferred) type is `'a list`, then I don't find it helpful to emphasize that the containing type is `list` by pluralizing the initial single letter with an `s` after it.  We are in the `List` module, so I think it is unnecessary to further emphasize that we are working with a `list`.

Instead, the letter `m` there is short for "monad", which I find is a more helpful reminder, especially in F# where we have to implement functions like functor's `map` for each functor because we lack higher kinded types.  There are many other monads that also have a `cons` function, and the current implementation is not inherently unique to `list` (....it is only syntactically unique).

Another advantage of the `ma` notation is that it looks like the generic type definition `M<'a>`.  If we were in some computational expression, then it is very natural to write `let! a = ma`.  This notation also scales in that it is also natural to write `mma` for an instance of type `M<M<'a>>` as well as `let! ma = mma` followed by `let! a = ma`.

I think it would be reasonable to consider a different letter than `m` since this arguments would also apply to (say) functors.  However, I don't like look/feel of `fa`, and this is also a bit confusing since `f` is most often used to identify a function.  I think using `c`, short for "container" would be reasonable.  In that case, the name would be `ca`.  I think that look good.

Reply:
I picked up this naming convention from [`language-ext`](https://github.com/louthy/language-ext).  I just did a search for `ma` (including matching case and whole word).  There are 29758 matching lines.  For `mma` there are 22 matching lines.

Reply:
> I prefer `a` over `x` since it matches the inferred type assigned by Visual Studio's IntelliSense and visible in the tooltips.

I don't find `x: 'a` to be a problem, but I'm fine with keeping `a`.

I don't find the monad references of the `m` prefix helpful. In F#, it's common to append `s` to signal that a variable is a collection, e.g. `x` and `xs`. I can't find this explicitly mentioned in official MS guides, but [this location](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-lists-and-arrays) names it like that, and other well-known resources also do that (e.g F# for fun and profit, if I remember correctly).

In any case, I won't loose any sleep over this. I'll leave it to you to find a good name (or keep the current one) and resolve this conversation.

Reply:
After all that, I just pushed 51ea9f210244c17e0d54773e46cc824cc7457d03 to `master`, which changes the names to `head` and `tail`.

I still like the other naming style, but I think these names are better in this case.

Thanks for suggesting that these names could be improved :)